src/global_scheduler/global_scheduler.h

```diff
@@ -73,6 +78,9 @@ typedef struct {
   aux_address_entry *photon_plasma_map;
   /** Objects cached by this global scheduler instance. */
   scheduler_object_info *scheduler_object_info_table;
+  /** An array of tasks that are currently impossible to schedule due to their
+   *  resource requirements. */
+  UT_array *impossible_tasks;
```

Comment:
I would consider calling it `pending_tasks_queue`. It is more general, and we may decide to queue tasks for performance reasons, not necessarily for resource constraint infeasibility reasons.